src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -74,4 +101,81 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
     protected Optional<TokenPermissions> getTokenPermissions(HttpServletRequest request) {
         return AuthorisationUtil.getTokenPermissions(request);
     }
+
+    private boolean doRoeUpdateJourneyChecks(HttpServletRequest request, HttpServletResponse response, String reqId, String companyNumberInScope,
+                                             String transactionId, Map<String, Object> authInfoMap) {
+        boolean isValidOeNumber = companyNumberInScope.startsWith("OE");
+        if (isValidOeNumber) {
+            boolean doCompanyNumbersMatch = doCompanyNumbersMatch(request, response, companyNumberInScope, transactionId);
+            authInfoMap.put("roe_update_journey_company_numbers_match", doCompanyNumbersMatch);
+            if (doCompanyNumbersMatch) {
+                ApiLogger.debugContext(reqId, "UserAuthenticationInterceptor authorised with company_oe_annual_update=create permission",
+                        authInfoMap);
+                return true;
+            } else {
+                ApiLogger.errorContext(reqId, "UserAuthenticationInterceptor unauthorised for ROE Update Journey as Company Numbers do not match", null, authInfoMap);
+                if (HttpServletResponse.SC_INTERNAL_SERVER_ERROR != response.getStatus()) {
```

Comment:
I think it is a bit dangerous to check that the response status is already set. How about doComanyNumberMatch always setting the response code? Then no need to back fill here.